### ISSUE TITLE
Implement expiration date for shares

### ISSUE THREAD

**Issue by PVince81** [owncloud/ocis#1250]
Expiration date of share is not yet supported in OCIS for any of the share types:
- user
- group
- public link

The API test scenarios for these are currently running in CI and are in the expected-failures file.

ToDo:
- [x] find all the share expiration scenarios and put them together in the expected-failures file, linked to this issue, and in sections for user, group, and public link shares #2412 
- [ ] implement in oCIS
- [ ] make it configurable in oCIS with similar options to oC10

This was the original post, preserved for posterity:
----------
As I can see in the code this is not implemented yet.

As for tests, it seems most will stay disabled because we need to set some settings with occ...

- [ ] tag the matching tests with this ticket
- [ ] implement expiration date for user shares
- [ ] somehow make it configurable as we have no occ in ocis yet
- [ ] enable tests

@pmaier1 FYI for adding to the roadmap somewhere. feel free to move to another project than MVP as this is blocked by the config stuff
@individual-it FYI

**Comment by phil-davis** [comment]
This issue specifically mentions expiration date for user shares. I think that we also want expiration dates for group shares and public link shares. I suspect that implementation for all of these will be done together.

@pmaier1 is this issue still current (or is there another issue somewhere about this)?

Do we expand this issue to also cover group shares an public link shares? Or create separate issues?

**Comment by phil-davis** [comment]
Assigned myslef so that I see this next week. This is a good example of a feature that is not implemented yet. Currently some of the related test scenarios are tagged in a way that they do not run in oCIS CI, and some are running in oCIS CI and listed in expected-failures. We need a consistent process for tagging tests.

**Comment by pmaier1** [comment]
> @pmaier1 is this issue still current (or is there another issue somewhere about this)?

This is still current but it is clearly a feature on the roadmap, not a bug. I'd be happy to exclude this feature from our hardening efforts and start working on it when the time has come.

**Comment by phil-davis** [comment]
@pmaier1 should I edit the title to be "Implement expiration date for all shares" and mention the various share types in the top ost?
(i.e. make this an issue about all expiration dates)

**Comment by pmaier1** [comment]
> @pmaier1 should I edit the title to be "Implement expiration date for all shares" and mention the various share types in the top ost?

If I'm not mistaken, the expiration date for links feature is already implemented. If that's not the case, I agree with your proposal.

**Comment by phil-davis** [comment]
Note:
`Scenario Outline: sharing with default expiration date enabled but not enforced for groups, user shares with expiration date`

That fails (and other group expiration date test scenarios)

`Scenario Outline: Create a public link with default expiration date set and max expiration date enforced and access using the public WebDAV API`

That also fails: `apiSharePublicLink1/createPublicLinkShare.feature:652` is in the expected failures file and I can see it failing in a recent drone build.

So share expiration is not yet working for any of the share type. I will adjust the top post.

**Comment by pmaier1** [comment]
These policies (enforce expiration date for user shares / public links) are also new features for oCIS - to be handled via the roadmap.

**Comment by phil-davis** [comment]
This issue is referenced in expected-failures. Reopening.

**Comment by ScharfViktor** [comment]
expiration date for shares works correct. I close issue
The tests were removed from the expected failure file and rewritten using e2e https://github.com/owncloud/web/pull/10292
